standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/FilterUtils.java

```diff
@@ -442,4 +447,35 @@ public static void checkDbAndTableFilters(boolean isFilterEnabled,
       return compactions;
     }
   }
+
+  public static GetLatestCompactionInfoResponse filterLatestCompactionInfoIfEnabled(
+      boolean isFilterEnabled, MetaStoreFilterHook filterHook,
+      String catName, String dbName, String tableName, GetLatestCompactionInfoResponse response
+  ) throws MetaException {
+    if (isFilterEnabled && response.getCompactionsSize() > 0) {
+      List<LatestCompactionInfo> compactions = response.getCompactions();
+      if (compactions.get(0).getPartitionname() == null) {
+        // non partitioned table
+        List<String> tableNames = new ArrayList<>();
+        tableNames.add(tableName);
+        tableNames = filterHook.filterTableNames(catName, dbName, tableNames);
+        if (tableNames.isEmpty()) {
+          response.unsetCompactions();
```

Comment:
Where is this implemented?

Reply:
This is a method that is auto-generated by Thrift for `GetLatestCompactionInfo`.